### PR TITLE
#14 Migrate Firestore to gitlive multiplatform

### PR DESCRIPTION
**Pull request by MarekMacko**

### CODE REVIEW

File: app/src/main/kotlin/com/intive/picover/common/di/AppModule.kt [hunk 1]
```diff
@@ -24,7 +24,7 @@ object AppModule {
 
 	@Provides
 	fun providePartiesRepository() =
-		PartiesRepository(Firebase.firestore)
+		PartiesRepository(dev.gitlive.firebase.Firebase.firestore)
```
Comment (MarekMacko) [inline app/src/main/kotlin/com/intive/picover/common/di/AppModule.kt:27]: full package needed due to mix with regular firebase package. To be simplified when `Firebase.storage.reference` will be migrated as well

File: gradle/libs.versions.toml [hunk 2]
```diff
@@ -31,7 +33,7 @@ firebase-bom = "com.google.firebase:firebase-bom:32.6.0"
 firebase-analytics = { module = "com.google.firebase:firebase-analytics-ktx" }
 firebase-crashlytics = { module = "com.google.firebase:firebase-crashlytics-ktx" }
 firebase-storage = { module = "com.google.firebase:firebase-storage-ktx" }
-firebase-firestore = { module = "com.google.firebase:firebase-firestore" }
+firebase-firestore = { module = "dev.gitlive:firebase-firestore", version.ref = "firebase" }
```
Comment (MarekMacko) [inline gradle/libs.versions.toml:36]: It's listed on `firebaseopensource` so it seems to be the best option at this moment. 
https://firebaseopensource.com/projects/gitliveapp/firebase-kotlin-sdk/ 
Github: https://github.com/gitliveapp/firebase-kotlin-sdk/

File: shared/build.gradle.kts [hunk 3]
```diff
@@ -10,7 +10,7 @@ kotlin {
 
 	sourceSets {
 		commonMain.dependencies {
-			implementation(project.dependencies.platform(libs.firebase.bom))
+			implementation(libs.coroutines)
```
Comment (MarekMacko) [inline shared/build.gradle.kts:13]: GitLive doesn't support bom

File: gradle/libs.versions.toml (hunk 2 above)
Comment (MarekMacko) [reply gradle/libs.versions.toml:36]: Upcoming migrations will use extracted version `firebase` so I've added it now. 

In the end, all of the dependencies will be migrated to `dev.gitlive` from `com.google.firebase`. 
Do we want to stick to `firebase` naming or we want to distinguish it, e.g by name `firebase-kmp`?

Comment (gmatyszczak) [reply gradle/libs.versions.toml:36]: Seems legit, I would keep it simply `firebase`

**Review by gmatyszczak — APPROVED** [review]

**Comment by MarekMacko** [comment]
@gmatyszczak Side note, issues have been revealed when configuration for new target has been added. In this case: https://github.com/Android-Guild/Picover-KMP/pull/19/files 
```
	listOf(
		iosX64(),
		iosArm64(),
		iosSimulatorArm64(),
	).forEach {
		it.binaries.framework {
			baseName = "shared"
			isStatic = true
		}
	}
```  
So the lesson here is: gradle configuration phase reveals errors when `common` source set dependency doesn't cover all configured targets. That's why there was no errors previously, we had only android target set, so we could put android specific libraries there. 

Above, of course is just a simplification because it depends what is actually used in shared code 😃

**Pull request merged**